sdk/src/types/block/payload/transaction/essence/regular.rs

```diff
@@ -419,6 +433,99 @@ fn verify_payload_packable<const VERIFY: bool>(
     Ok(())
 }
 
+#[derive(Copy, Clone, Eq, PartialEq, Debug)]
+#[non_exhaustive]
+#[repr(u8)]
+pub enum TransactionCapabilityFlag {
+    BurnNativeTokens = 0b00000001,
+    BurnMana = 0b00000010,
+    DestroyAccountOutputs = 0b00000100,
+    DestroyFoundryOutputs = 0b00001000,
+    DestroyNftOutputs = 0b00010000,
+}
+
+impl TransactionCapabilityFlag {
+    pub fn all() -> impl Iterator<Item = Self> {
+        [
+            Self::BurnNativeTokens,
+            Self::BurnMana,
+            Self::DestroyAccountOutputs,
+            Self::DestroyFoundryOutputs,
+            Self::DestroyNftOutputs,
+        ]
+        .into_iter()
+    }
+}
+
+#[derive(Copy, Clone, Debug, Default, Eq, PartialEq, Ord, PartialOrd, Hash, Deref, Packable)]
+#[repr(transparent)]
+pub struct TransactionCapabilities(u8);
```

Comment:
Will you replicate the address flags way? This should also be variable

Reply:
It is?? I thought this was a single byte :C

Reply:
Yeah it is, no TIP yet so it's not clear sorry. I've just asked Phillip and Max and they confirmed.

Reply:
That's a bummer, but at least we can share types then. I'll draft this until that one is merged

Reply:
No, the flags aren't but the array type can be shared